### ISSUE TITLE
Doesn't work with a new repo when using npm [turborepo] 

### ISSUE THREAD

**Issue by AmirL** [vercel/turbo#5331]
### What version of Turborepo are you using?

1.10.3

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Mac

### Describe the Bug

```
npm run dev

> dev
> turbo run dev

• Running dev
• Remote caching disabled
**root task dev (turbo run dev) looks like it invokes turbo and might cause a loop**

No tasks were executed as part of this run.

 Tasks:    0 successful, 0 total
Cached:    0 cached, 0 total
  Time:    739ms

 ERROR  run failed: command  exited (1)
```

### Expected Behavior

Obviously, a freshly generated project should work after "npm run dev" or any other npm or turbo command. 

I checked with pnpm - and it works well with it. 

### To Reproduce

- npx create-turbo@latest 
- select npm as a package manager
- go into the project dir and run "npm run dev" (or "npx turbo run dev")

**Comment by chris-olszewski** [comment]
Hi @AmirL, could you provide the `package.json` that got generated? I tried to reproduce following your steps and was unable to.

**Comment by AmirL** [comment]
https://github.com/AmirL/turborepo-bug-test

I tried it on codesandbox and it works. But it doesn't work on my mac. 

node version v20.2.0
npm version 8.19.2

PS Also, there is another case with the same problem:
https://github.com/vercel/turbo/discussions/5290

**Comment by gabriellet** [comment]
I have this problem on a Mac using yarn with versions 1.10.3 and 1.10.6. I don't see the issue on 1.10.4 though.

**Comment by eykrehbein** [comment]
Also breaks for me with yarn `1.22.19` and turbo > `1.10.4` . As @gabriellet, pinning turbo `1.10.4` works

**Comment by marlonmarcello** [comment]
I am experiencing the same issue:
```
node --version
v18.16.1

yarn --version
1.22.19

turbo --version
1.10.8
```

**Comment by sammyfilly** [comment]
Request bot : to return 
Return errors : import from turbo/Vercel errors 
 
 throw errors: echo tools/ways to resolve issues 
Import resolvers

-inspect: resolver’s resolving issues 
-Create new files for Resolved issues 
-throw resolved issues back to code
-fix issues

**Comment by kkan0615** [comment]
any solutions?

**Comment by jmcunningham** [comment]
I'm seeing this issue on my mac, using the "latest" turbo in package.json.

I started with the basic example project, then I renamed the "web" app to something else and that's when it broke for me with the error listed above.

Per other comments, I tried reverting back to Turbo 1.10.4 and the issue went away.

Interesting...I then set turbo back to "latest" in my root package.json and reinstalled turbo (which appears to be 1.10.9) and it works. So reverting fixed it, then enabled me to go back to latest? As if the latest version doesn't like changes to a top-level within the apps directory.

**Comment by robcrockhlx** [comment]
> Also breaks for me with yarn `1.22.19` and turbo > `1.10.4` . As @gabriellet, pinning turbo `1.10.4` works

This worked for me too, but it is concerning that breaking changes like this have been introduced 😬

**Comment by vana-dev** [comment]
same problem

**Comment by zifahm** [comment]
having the same issue! not npm tho, using yarn berry.

**Comment by GhostWalker562** [comment]
Same issue here with 
"packageManager": "pnpm@8.6.9",
and 
"turbo": "^1.10.12",

**Comment by JulesVision** [comment]
Same issue with :
"packageManager": "yarn@1.22.17",
and
"turbo": "^1.10.12"

**Comment by JulesVision** [comment]
I found a workaround. Run each app separately with :
`yarn dev --filter [app_name]`
And after run all the apps with `yarn dev`

**Comment by angelzou** [comment]
> Same issue with : "packageManager": "yarn@1.22.17", and "turbo": "^1.10.12"

same issue with:
yarn: 1.22.19
turbo: 1.10.12

**Comment by booi-dev** [comment]
I have the same issue.... ONLY WHEN I added 'turbo build' in the script. Idk, it create kind of conflict. I remove that, and run turbo build without the script. And it work. 

Turbo time.

**Comment by Mistat** [comment]
I have same issue. after 1.10.5  
1.10.4 is fine

**Comment by vagasrh** [comment]
Same issue with :
"packageManager": "yarn@1.22.17",
and
"turbo": "^1.10.12"

**Comment by uncleSlayer** [comment]
> I'm seeing this issue on my mac, using the "latest" turbo in package.json.
> 
> I started with the basic example project, then I renamed the "web" app to something else and that's when it broke for me with the error listed above.
> 
> Per other comments, I tried reverting back to Turbo 1.10.4 and the issue went away.
> 
> Interesting...I then set turbo back to "latest" in my root package.json and reinstalled turbo (which appears to be 1.10.9) and it works. So reverting fixed it, then enabled me to go back to latest? As if the latest version doesn't like changes to a top-level within the apps directory.
> 
> Edit: If it helps, my packageManger is "npm@8.5.5".

this beautifully worked for me too with yarn. Thanks

**Comment by zhanghengxin** [comment]
change your version.

eg,1.9.1

**Comment by joaom182** [comment]
The solution for now is to fix the turbo package version to `1.10.4`

```
{
  "devDependencies": {
    "turbo": "1.10.4"
  }
}
```

**Comment by blockmood** [comment]
```
• Running start
• Remote caching disabled

No tasks were executed as part of this run.

 Tasks:    0 successful, 0 total
Cached:    0 cached, 0 total
  Time:    995ms
```

**Comment by revoltez** [comment]
newbie advice: in my case i forgot to run `pnpm(or npm or yarn) install` first to install the dependencies, rerun `turbo dev` and it works

**Comment by swalker326** [comment]
I was hitting this issue and didn't have a root turbo.json, in my case I was on docker and hadn't copied it over. It didn't throw an error but the `pnpm build` command called `turbo build` which threw the above error. Adding `COPY turbo.json ./` resolved my issue

**Comment by thertzelle** [comment]
> I have same issue. after 1.10.5 1.10.4 is fine

Same here; Using 1.10.4 is fine, but after 1.10.5 turbo repo gives me the loop error. "looks like it invokes turbo and might cause a loop"

**Comment by SkylerRogers** [comment]
+1 - weirdest part about this is if I re-clone my repo using latest version of turbo, the turbo builds start to work again. Can't figure out why...

**Comment by mehulkar** [comment]
> https://github.com/AmirL/turborepo-bug-test
> 
> I tried it on codesandbox and it works. But it doesn't work on my mac.
> 
> node version v20.2.0 npm version 8.19.2
> 
> PS Also, there is another case with the same problem: #5290

Thanks for the repro @AmirL, I'm not able to reproduce with this config. Any chance there's something else going on with your local setup? The other comments on this repo seem varied and suggest different problems, but if you're able to reproduce with this sample repo without any changes, I'd love to see `turbo dev -vvv` logs to figure out if this an issue with global+local version mismatches or what. I tried pinning your reproduction's `turbo` in the root package.json to `1.10.13` to reproduce and also tried with the existing `latest` config (which is now `1.10.14`).

**Comment by mehulkar** [comment]
> I have the same issue.... ONLY WHEN I added 'turbo build' in the script. Idk, it create kind of conflict. I remove that, and run turbo build without the script. And it work.
> 
@booi-dev this may or may not be related, you should open another issue with your `turbo.json`.

**Comment by nicolas-angelo** [comment]
Just a heads up, I had this same issue/same error in a yarn workspaces monorepo - 
turns out I forgot to add "version" in the root package.json. 
After that, all my turbo commands worked

**Comment by mfnash** [comment]
Same here, I'm using pnpm ver 8.8.0 and node ver 18. I've already downgrade turbo to 1.10.4 and 1.9.1 but it still not work. Can't figure out why...

**Comment by JLarky** [comment]
I got this error when I was in a yarn project but it got package-lock.json file in it by accident. Removing extra lock file helped

**Comment by romeerez** [comment]
I was using package.json5 - works for pnpm, but turbo doesn't see them, had to change them to json.

**Comment by luke-quest** [comment]
I had two lock files installed with pnpm.

Downgrading to tubo 1.10.4 made the project work and returned a warning about two lock files. After deleting the second lock file, I could update back to the latest turbo version and it worked.

**Comment by jeffreys-cat** [comment]
╭───────────────────────────────────────────────────────────────────────╮
│                                                                       │
│                  Update available v1.10.12 ≫ v1.13.0                  │
│    Changelog: https://github.com/vercel/turbo/releases/tag/v1.13.0    │
│               Run "npx @turbo/codemod update" to update               │
│                                                                       │
│     Follow @turborepo for updates: https://twitter.com/turborepo      │
╰───────────────────────────────────────────────────────────────────────╯
• Running build:webui:byoc:zh_CN
• Remote caching disabled

No tasks were executed as part of this run.

 Tasks:    0 successful, 0 total
Cached:    0 cached, 0 total
  Time:    21ms

**Comment by anthonyshew** [comment]
Hey, folks. I'm seeing a lot of different types of reports on this issue. I'm seeing everything from using invalid `.json` extensions to multiple lockfiles to missing dependencies - and much more.

We want to keep each individual issue focused *just one actionable thing for Turborepo*. In the interest of that, I'm going to close and lock this issue. if you have an issue report that you'd like to put on the team's radar, please feel free to open an issue with a reproduction so we can investigate.
